airbyte-integrations/connectors/source-exchange-rate-api/source_exchange_rate_api/source.py

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+import datetime
+from abc import ABC
+from typing import Any, Iterable, List, Mapping, MutableMapping, Optional, Tuple
+
+import requests
+
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream, IncrementalMixin
+from airbyte_cdk.sources.streams.http import HttpStream
+
+from source_exchange_rate_api.authenticator import HTTPApiKeyHeaderAuth
+
+
+class ExchangeRateApiStream(HttpStream, ABC):
+    url_base = "https://api.apilayer.com/exchangerates_data/"
+
+    def __init__(self, config: Mapping[str, Any], *args, **kwargs):
+        super(ExchangeRateApiStream, self).__init__(*args, **kwargs)
+        self.config: Mapping[str, Any] = config
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield response.json()
+
+
+class LatestRates(ExchangeRateApiStream):
+    primary_key = "timestamp"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "latest"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"base": self.config["base"]}
+        if "symbols" in self.config and self.config["symbols"]:
+            params["symbols"] = self.config["symbols"]
+        return params
+
+
+class IncrementalExchangeRateApiStream(HttpStream, IncrementalMixin, ABC):
```

Comment:
Incremental streams should always have a cursor field. I suggest you define it here or explicitly mark it abstract

Reply:
It is ABC class. Сursor field is defined in concrete successor class (HistoricalRates) because each steam may have its own cursor field. Please correct if I understood it wrong

Reply:
that's right but we can add an abstract cursor here to make it more obvious